tests/client/test_call_graph.cpp

```diff
@@ -0,0 +1,233 @@
+    std::queue<std::shared_ptr<CallGraph::TreeNode>> expected_fnc_queue;
+    actual_fnc_queue.push(actual.first); 
+    expected_fnc_queue.push(expected.first);
+
+    while (!actual_fnc_queue.empty())
+    {
+        auto& actual_fnc = actual_fnc_queue.front();
+        auto& expected_fnc = expected_fnc_queue.front();
+
+        if (expected_fnc->data.label != actual_fnc->data.label)
+        {
+            return ::testing::AssertionFailure() << std::endl 
+                << "Expected label: " << expected_fnc->data.label << std::endl
+                << "Actual label: " << actual_fnc->data.label << std::endl;
+
+        }
+        if (expected_fnc->data.start_ts != actual_fnc->data.start_ts)
+        {
+            return ::testing::AssertionFailure() << std::endl 
+                << "For node with label: " << expected_fnc->data.label << std::endl
+                << "Expected start_ts: " << expected_fnc->data.start_ts << std::endl
+                << "Actual start_ts: " << actual_fnc->data.start_ts << std::endl;
+        }
+        if (expected_fnc->data.stop_ts  != actual_fnc->data.stop_ts)
+        {
+            return ::testing::AssertionFailure() << std::endl 
+                << "For node with label: " << expected_fnc->data.label << std::endl
+                << "Expected stop_ts: " << expected_fnc->data.stop_ts << std::endl
+                << "Actual stop_ts: " << actual_fnc->data.stop_ts << std::endl;
+        }
+        
+        if (expected_fnc->children.size() != actual_fnc->children.size())
+        {
+            return ::testing::AssertionFailure() << std::endl 
+                << "For node with label: " << expected_fnc->data.label << std::endl
+                << "Expected number of children: " << expected_fnc->children.size() << std::endl
+                << "Actual number of children: " << actual_fnc->children.size() << std::endl;
+        }
+
+        for (size_t i = 0; i < actual_fnc->children.size(); ++i)
+        {
+            if (expected_fnc->children[i].second != actual_fnc->children[i].second) 
+            {
+                return ::testing::AssertionFailure() << std::endl
+                    << "For node with label: " << expected_fnc->data.label << std::endl
+                    << "Expected number of calls to a child: " << expected_fnc->children[i].second << std::endl
+                    << "Actual number of calls to a child: " << actual_fnc->children[i].second << std::endl;
+            }
+
+            actual_fnc_queue.push(actual_fnc->children[i].first);
+            expected_fnc_queue.push(expected_fnc->children[i].first);
+        }
+
+        actual_fnc_queue.pop();
+        expected_fnc_queue.pop();
+    }
+
+    return ::testing::AssertionSuccess();
+}
+
+void init(std::vector<CallGraph::NodeData>& events,
+        std::vector<std::pair<std::shared_ptr<CallGraph::TreeNode>, int>>& tree,
+        std::string file_name)
+{
+    std::ifstream file;
```

Comment:
This function is quite huge, so I think it'd be better to split it to multiple smaller functions. Also, you might consider creating a class which implements the functionality (test tree loader).